### DISCUSSION TITLE
Fan Curve Control

### DISCUSSION THREAD

**Discussion by wearver** [T-Troll/alienfx-tools#395] [General]
I reported a situation previously and I would like to know if it has been resolved in these new versions or if it is a feature of notebook management.
I have a G15 Ryzen Edition and the Fan Curve works perfectly, however, when the CPU and GPU temperature reaches a certain level, the system takes control of the speeds and only releases it when the temperature returns to a pre-defined level. Therefore, only in certain situations can I control the speed.

**Comment by T-Troll** [comment]
It's correct. 

At this hardware, fan control is indirect, so you can't override BIOS settings, but can increase fan speeds in some situations. Boost is not affect fan RPMs directly, but give BIOS a hint to increase it.

In case you will be a bit clear about what do you want to archive, i may give you some hints.

**Reply by wearver** [reply]
I would like to configure the fans to a certain temperature, but whenever it heats up and deviates from the ramp I configured, the bios takes over.

**Reply by T-Troll** [reply]
Again, fan control is INDIRECT, there is no way to set exact fan speed. All we can do to get BIOS the hint how to do it.

**Comment by Wromper** [comment]
Following up on this, if I set the fan boost to 0, it seems like it should prevent the fans from going to 100%.  But in my laptop, even low temps and a fan boost of 0, still gets fans to 80%-100% and temps 70C to 80C.  Trying to keep the fans tame because the laptop isnt getting hot, just the fans spin too fast.  I wish there was a setting to override the fan command and force under a certain % or RPM.  Like above, I really dont care about the exact RPM, but I do care how loud it gets unnecessarily.  Any ideas ... would love love to force the fans to stay under 70%.  thanks.

**Reply by T-Troll** [reply]
It's correct. In case you stop/slow fans at so HIGH temperature between 70 and 80, first load spike bring CPU to throttling. Fans are quite inertial...

**Reply by Wromper** [reply]
I understand that stopping the fans can cause temps to spike quickly, I just want the fans to stop at 70%-80% RPM instead of 100% RPM.  There isnt much difference in cooling between 80% and 100%.  I have closely monitored my temps and they dont come close to 100C.  Also, I have the 7845HX, which is 12 cores.  I dont need 12 cores for gaming, so I went into the BIOS and set it to use 8 cores, which is 16 threads.  With that, the CPU now uses 33% less power (I verified).  Then I ran CPU-Z and Cinebench multi-thread CPU tests.  Even at 100% CPU util, the temps are still around 80C or below.  That's my issue, there is no way for the CPU to overheat even with the fans at 60% RPM.  I am getting kinda deperate to keep the noise down, any thoughts?
thanks again.

**Reply by wearver** [reply]
I was looking for articles about a HomeAssistant project that I'm setting up and I came across this project. It is an access to the bios where you can change the parameters of how the system will control the fans. I haven't tested the parameters yet, but I went into the menus and there's a lot. T-Troll, can you change alienfx to have direct access to the hardware?

https://github.com/DavidS95/Smokeless_UMAF?tab=readme-ov-file

**Reply by wearver** [reply]
> I understand that stopping the fans can cause temps to spike quickly, I just want the fans to stop at 70%-80% RPM instead of 100% RPM. There isnt much difference in cooling between 80% and 100%. I have closely monitored my temps and they dont come close to 100C. Also, I have the 7845HX, which is 12 cores. I dont need 12 cores for gaming, so I went into the BIOS and set it to use 8 cores, which is 16 threads. With that, the CPU now uses 33% less power (I verified). Then I ran CPU-Z and Cinebench multi-thread CPU tests. Even at 100% CPU util, the temps are still around 80C or below. That's my issue, there is no way for the CPU to overheat even with the fans at 60% RPM. I am getting kinda deperate to keep the noise down, any thoughts? thanks again.


The Cpu controls clocks to avoid heating. Even with the fan at low rpm, it will not exceed a certain temperature, as it will reduce clock speeds. I use the Ryzen controller which you can put your own parameters into.

**Reply by T-Troll** [reply]
Oh, i know this guy.
So... MOST of Alienware models after m15R1 does not have direct fan control via EC.
Yes, playing with BIOS (modify settings or make custom one) can alter PASV/HOT points and (maybe, but hard to find) curves, but for the price of the risk to burn CPU or brick entire system.
You can take a risk, but i will not provide it for the users of my tools.

**Reply by wearver** [reply]
> Oh, i know this guy. So... MOST of Alienware models after m15R1 does not have direct fan control via EC. Yes, playing with BIOS (modify settings or make custom one) can alter PASV/HOT points and (maybe, but hard to find) curves, but for the price of the risk to burn CPU or brick entire system. You can take a risk, but i will not provide it for the users of my tools.

I didn't make this change precisely because of the risk it brings. According to the author himself, any incorrect change could damage the bios, if it were a bios similar to motherboards, which supports updating without having to enter it, that would be fine. I saw some people saying that Dell offers updates in these cases, do you know anything?
